chronos_bin/src/message_processor.rs

```diff
@@ -14,157 +15,157 @@ pub struct MessageProcessor {
-                        for row in publish_rows {
-                            let updated_row = TableRow {
-                                id: row.get("id"),
-                                deadline: row.get("deadline"),
-                                readied_at: row.get("readied_at"),
-                                readied_by: row.get("readied_by"),
-                                message_headers: row.get("message_headers"),
-                                message_key: row.get("message_key"),
-                                message_value: row.get("message_value"),
-                            };
-                            let mut headers: HashMap<String, String> = match serde_json::from_str(&updated_row.message_headers.to_string()) {
-                                Ok(t) => t,
-                                Err(_e) => {
-                                    println!("error occurred while parsing");
-                                    HashMap::new()
-                                }
-                            };
-                            //TODO: handle empty headers
-
-                            let readied_by = updated_row.readied_by.to_string();
-                            tracing::Span::current().record("node_id", &readied_by);
-                            headers.insert("readied_by".to_string(), readied_by);
-
-                            tracing::Span::current().record("correlationId", updated_row.id.to_string());
-
-                            publish_futures.push(self.producer.kafka_publish(
-                                updated_row.message_value.to_string(),
-                                Some(headers),
-                                updated_row.message_key.to_string(),
-                                // updated_row.id.to_string(),
-                            ))
-                        }
                         let results = futures::future::join_all(publish_futures).await;
-                        for result in results {
-                            match result {
-                                Ok(m) => {
-                                    tracing::Span::current().record("is_published", "true");
-                                    ids.push(m);
-                                }
-                                Err(e) => {
-                                    tracing::Span::current().record("is_published", "false");
-                                    tracing::Span::current().record("error", &e.to_string());
-
-                                    log::error!("Error: delayed message publish failed {:?}", e);
-                                    break;
-                                    // failure detection needs to pick
-                                }
-                            }
-                        }
+
+                        let ids: Vec<String> = results.into_iter().map(Self::gather_ids).collect();
```

Comment:
Please re-visit this part. I am not sure the consequence of having `gather_ids` to return an empty string. Just by reading the code I see that empty string will then be fed into data store delete method.

I would suggest to think if your results list is allowed to have any Result::Err entries. 

- If no errors are allowed then you want to validate the list and abort with error message composed of all those errors. 
- If there could be some errors mixed with Result::Ok entries then you better filter errors by using `filter()` function on the iterator. And then process only Results::Ok. However, in this scenario I don't know what to do with filtered out errors. It depends on requirements. The current code appears to only log errors but do nothing with faulty entries (just taking an empty string instead of ID).